### PR TITLE
Fix headers of article list & smart folder localization and behavior

### PR DESCRIPTION
**Pull request by barijaona**
Closes issues #933 and #944 : 
- change localizable strings to appear directly in source code
- change test of end of criteria when building the popup menu

### CODE REVIEW

File: src/Criteria.m [hunk 1]
```diff
@@ -238,6 +238,18 @@ +(CriteriaCondition)conditionFromString:(NSString *)string
...
 
+/*
+ the following  code NEVER runs
+ it is only there to ease localization export through Xcode
+*/
+-(void)CriteriaConditionLocalization
+{
+    NSString * dummy;
+#pragma unused(dummy)
+    dummy = NSLocalizedString(@"any", "rule for association of multiple criteria");
```
Comment (Eitot) [inline src/Criteria.m]: Add `@`.

File: src/Criteria.m [hunk 2]
```diff
@@ -238,6 +238,18 @@ +(CriteriaCondition)conditionFromString:(NSString *)string
...
+    dummy = NSLocalizedString(@"all", "rule for association of multiple criteria");
```
Comment (Eitot) [inline src/Criteria.m]: Add `@`.

**Comment by barijaona** [comment]
In fact, the -operatorFromString: method was not used.
I converted to -operatorFromLocalizedString just as a matter of precaution, but if you prefer, one can get rid of it.

**Comment by josh64x2** [comment]
I would prefer we use this as an opportunity to come up with a correct long term solution (perhaps using NSPredicateEditor or NSRuleEditor as @Eitot suggested). 

I'd suggest putting this PR on hold and investigate if it is feasible to implement the other possible solutions.

**Comment by barijaona** [comment]
I agree this section of code is old and I'd encourage efforts on designing something new, but I do thing we really need to merge this PR in between. Because : 
- this would fullfill the migration of the translations project to Crowdin and constitute a stable basis on which many more people than the three of us could be involved,
- after so many changes, we need some beta-testing. But it would hurt the morale of testers if we deliver a version with functional regressions, while short term solutions are available (and can easily be scratched later)

May I repeat my open source mantra ? Commit as soon as possible, deliver early and get feedback.
Or if word it differently : make your users happy 😊, today whenever possible 😀🤣.

**Comment by barijaona** [comment]
I'd add that even with a NSPredicate solution, the localization of the fields name would still be useful and is also required for main menu and article list.

**Comment by josh64x2** [comment]
@barijaona please don't merge your own changes that haven't been approved in future.

**Pull request merged**